lib/vast/Conversion/ABI/EmitABI.cpp

```diff
@@ -117,9 +117,18 @@ namespace vast
                 return out;
             }
 
-            core::FunctionType abified_type() {
-                // TODO: deal with varaargs
-                return core::FunctionType::get(abified_args(), abified_rets());
+            core::FunctionType abified_type()
+            {
+                auto add_lvalue = [&](const auto &from)
```

Comment:
this might be nicer with `std::views::transform`, but I am not sure it will cooperate with MLIR Type constructors.